lib/common/output.c

```diff
@@ -32,12 +32,18 @@ pcmk__output_free(pcmk__output_t *out, crm_exit_t exit_status) {
 int
 pcmk__output_new(pcmk__output_t **out, const char *fmt_name, const char *filename,
                  char **argv) {
-    pcmk__output_factory_t create = NULL;;
+    pcmk__output_factory_t create = NULL;
 
     if (formatters == NULL) {
         return EINVAL;
     }
 
+    if (fmt_name != NULL && g_hash_table_contains(formatters, fmt_name) == false) {
+        return pcmk_err_unknown_format;
+    } else if (fmt_name == NULL && g_hash_table_contains(formatters, "text") == false) {
+        return pcmk_err_unknown_format;
```

Comment:
Sorry, we can't use g_hash_table_contains() ... it was added in glib 2.32, and we only require 2.16 at the moment. Otherwise I think this PR is ready.

Eventually I would like to have some sort of standard policy regarding minimum dependency versions. Right now it's a very rough "about 7-10 years from release" guideline. The most important thing is to support whatever releases common Linux distributions have to package pacemaker for, and beyond that, the hope would be to support what 90+% of users that compile pacemaker themselves are building on. Without an easy way to gather that info, the 7-10 guideline seems reasonably generous.

Reply:
After testing without this patch again, I'm not sure why I ever thought it was necessary.  crm_mon and stonith_admin work fine without it.  I don't recall the exact circumstances that led me to write it, but I am thinking perhaps it was more due to how code was arranged in the tools instead of the user passing bad data on the command line.  So this patch can probably just go away.

Reply:
Agreed, we can just drop this commit